### PR TITLE
Sparkconnector

### PR DESCRIPTION
**Pull request by diocas**

### CODE REVIEW

File: SwanPortAllocator/swanportallocator/__init__.py [hunk 1]
```diff
@@ -4,7 +4,7 @@ def _jupyter_nbextension_paths():
     # Empty to avoid error when automatically trying to enable all nbextensions
     return []
 
-def _jupyter_server_extension_paths():
+def _jupyter_server_extension_points():
```
Comment (etejedor) [inline SwanPortAllocator/swanportallocator/__init__.py:7]: Does this change match the commit message?

File: SparkConnector/README.md [hunk 2]
```diff
@@ -2,83 +2,92 @@
...
 
-Note: You will need NodeJS to install the extension.
+To install the extension, execute:
 
 ```bash
 pip install sparkconnector
-jupyter nbextension install sparkconnector --py
-jupyter nbextension enable  sparkconnector --py
-jupyter lab build
+jupyter nbclassic-extension install sparkconnector --py
```
Comment (etejedor) [inline SparkConnector/README.md:25]: Is @PMax5 already using this new way of installing in his Alma9 image? Just making sure.

File: SparkConnector/setup.py [hunk 3]
```diff
@@ -63,7 +63,7 @@
     cmdclass= cmdclass,
     packages=setuptools.find_packages(),
     install_requires=[
-        "jupyterlab~=3.0",
+        "jupyterlab>=4.0.0,<5",
```
Comment (etejedor) [inline SparkConnector/setup.py:66]: Here JupyterLab is required to install and to build?

File: SwanPortAllocator/swanportallocator/__init__.py (hunk 1 above)
Comment (diocas) [reply SwanPortAllocator/swanportallocator/__init__.py:7]: Yes.

File: SparkConnector/README.md (hunk 2 above)
Comment (diocas) [reply SparkConnector/README.md:25]: Not yet. I still need to review that part as I hope many of those can be (and are already) automated. And we need to check if we need to disable some as well (i.e the spark connector for lab should not load automatically as well).

File: SparkConnector/setup.py (hunk 3 above)
Comment (diocas) [reply SparkConnector/setup.py:66]: I'm not understanding the question.

Comment (diocas) [reply SparkConnector/setup.py:66]: Ah sorry, yes, you need lab to build things as well. Don't remember why, but that is part of their cookiecutter recipe.

**Review by etejedor — APPROVED** [review]

**Review by PMax5 — APPROVED** [review]

**Pull request merged**